svgelements/svgelements.py

```diff
@@ -5497,10 +5497,10 @@ def _validate_subpath(self, index):
             if isinstance(close_search, Close):
                 for k in range(index, -1, -1):
                     move_search = self._segments[k]
-                    if isinstance(move_search, Move):
+                    if isinstance(move_search, (Move, Close)):
                         self._segments[j].end = Point(move_search.end)
                         return
-                self._segments[j].end = Point(self._segments[0].end)
+                self._segments[j].end = Point(self._segments[0].start)
```

Comment:
.start for a Move or close cannot be correct here. The destination of the move is the move position, the start is where the path was before that, usually the previous subpath or None. The source of the Close is the the position where Close was called, the `end` is where you're going which is the destination of the Move.